question-servers/freeform.js

```diff
@@ -1173,13 +1190,47 @@ module.exports = {
       (cachedData, cacheHit) => {
         // function to process the cachedData, whether we
         // just rendered it or whether it came from cache
-        const { courseIssues, html, renderedElementNames } = cachedData;
-        callback(null, courseIssues, html, renderedElementNames, cacheHit);
+        callback(null, {
+          ...cachedData,
+          cacheHit,
+        });
       },
       callback // error-handling function
     );
   },
 
+  renderPanelInstrumented: async function (
+    panel,
+    pc,
+    submission,
+    variant,
+    question,
+    course,
+    locals,
+    context
+  ) {
+    return instrumented(`freeform.renderPanel:${panel}`, async (span) => {
```

Comment:
Currently, I only have instrumentation for renders, not any of the other operations we have (`generate`, `parse`, `grade`, etc.). Adding instrumentation has proven to be easiest with Promise-based functions, and I didn't want this PR to explode into a full-blown refactor of this code. I intend to do that in a separate PR and then come back to add instrumentation to `processQuestion`, which should give us much better visibility than just instrumenting renders.